ui/src/containers/NavigationModalContainer.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { Button } from '@patternfly/react-core';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { toggleNavigationModal, toggleAppDetailedIsDirty } from '../actions/actions-ui';
+import { LargeModal } from '../components/common/LargeModal';
+
+class NavigationModalContainer extends React.Component {
+  handleModalClose = () => {
+    this.props.toggleNavigationModal(false);
+  };
+
+  handleLeaveClick = () => {
+    this.props.toggleAppDetailedIsDirty();
+    this.props.history.goBack();
+    this.handleModalClose();
+  };
+
+  render () {
+    return (
+      <LargeModal
+        title="Are you sure you want to leave this page?"
+        isOpen={this.props.isNavigationModalOpen}
+        onClose={this.handleModalClose}
+        actions={[
+          <Button key="leave" variant="danger" onClick={this.handleLeaveClick}>
```

Comment:
Is red the colour we want to go with here? For me, red is reserved for extremely risky actions, such as deleting an app. Yes disabling an app is somewhat risky but is to the same degree?

I'm not sure myself by the way, it is just worth discussing.

Reply:
I decided to go with red because the user's unsaved data will be lost if they navigate away.